framework/PostProcessorFunctions/HStoPSoperator.py

```diff
@@ -0,0 +1,151 @@
+import warnings
+warnings.simplefilter('default',DeprecationWarning)
+if not 'xrange' in dir(__builtins__):
+  xrange = range
+#End compatibility block for Python 3----------------------------------------------------------------
+
+#External Modules------------------------------------------------------------------------------------
+import os
+import copy
+import numpy as np
+#External Modules End--------------------------------------------------------------------------------
+
+from PostProcessorInterfaceBaseClass import PostProcessorInterfaceBase
+
+
+class HStoPSoperator(PostProcessorInterfaceBase):
+  """
+   This Post-Processor performs the conversion from HistorySet to PointSet
+   The conversion is performed based on any of the following operations:
+   - row value
+   - pivot value
+   - operator
+  """
+
+  def initialize(self):
+    """
+     Method to initialize the Interfaced Post-processor
+     @ In, None
+     @ Out, None
+    """
+    PostProcessorInterfaceBase.initialize(self)
+    self.inputFormat  = 'HistorySet'
+    self.outputFormat = 'PointSet'
+    #pivotParameter identify the ID of the temporal variable in the data set based on which
+    # the operations are performed. Optional (defaul=time)
+    self.pivotParameter = 'time'
+    self.settings       = {'operationType':None,'operationValue':None,'pivotStrategy':'nearest'}
+
+  def readMoreXML(self,xmlNode):
+    """
+      Function that reads elements this post-processor will use
+      @ In, xmlNode, ElementTree, Xml element node
+      @ Out, None
+    """
+    foundPivot = False
+    for child in xmlNode:
+      if child.tag  == 'pivotParameter':
+        foundPivot, self.pivotParameter = True,child.text.strip()
+      elif child.tag in ['row','pivotValue','operator']:
+        self.settings['operationType'] = child.tag
+        self.settings['operationValue'] = float(child.text) if child.tag != 'operator' else child.text
+      elif child.tag  == 'pivotStrategy':
+        self.settings[child.tag] = child.text.strip()
+        if child.text not in ['nearest','floor','ceiling','interpolate']:
+          self.raiseAnError(IOError, '"pivotStrategy" can be only "nearest","floor","ceiling" or "interpolate"!')
+      elif child.tag !='method':
+        self.raiseAnError(IOError, 'XML node ' + str(child.tag) + ' is not recognized')
+    if not foundPivot:
+      self.raiseAWarning('"pivotParameter" is not inputted! Default is "'+ self.pivotParameter +'"!')
+    if self.settings['operationType'] is None:
+      self.raiseAnError(IOError, 'No operation has been inputted!')
+    if self.settings['operationType'] == 'operator' and self.settings['operationValue'] not in ['max','min','average']:
+      self.raiseAnError(IOError, '"operator" can be either "max", "min" or "average"!')
+
+  def run(self,inputDic):
+    """
+    This method performs the actual transformation of the data object from history set to point set
```

Comment:
add two spaces at the beginning of this line

Reply:
done